# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/audioipc/src/cmsghdr.c
# Commit: 634412ce2943
# Full Hash: 634412ce29430557eec6d0c0313f1908e61ad13d
# Author: Matthew Gregan <kinetik@flim.org>
# Date: 2021-08-11 03:46:10
# Regressor Bug: 1724141
# File Overlap Count: 6
# Description:
#   Bug 1724141 - Update audioipc to 8bb1a227.  r=chunmin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D121826
# ==============================================================================

diff -r 722284bed619 -r 634412ce2943 third_party/rust/audioipc/src/cmsghdr.c
--- a/third_party/rust/audioipc/src/cmsghdr.c	Tue Aug 10 20:38:51 2021 +0000
+++ b/third_party/rust/audioipc/src/cmsghdr.c	Tue Aug 10 20:52:03 2021 +0000
@@ -5,18 +5,18 @@
 const uint8_t*
 cmsghdr_bytes(size_t* size)
 {
-  int myfds[3] = {0, 1, 2};
+  int myfd = 0;
 
   static union {
-    uint8_t buf[CMSG_SPACE(sizeof(myfds))];
+    uint8_t buf[CMSG_SPACE(sizeof(myfd))];
     struct cmsghdr align;
   } u;
 
-  u.align.cmsg_len = CMSG_LEN(sizeof(myfds));
+  u.align.cmsg_len = CMSG_LEN(sizeof(myfd));
   u.align.cmsg_level = SOL_SOCKET;
   u.align.cmsg_type = SCM_RIGHTS;
 
-  memcpy(CMSG_DATA(&u.align), myfds, sizeof(myfds));
+  memcpy(CMSG_DATA(&u.align), &myfd, sizeof(myfd));
 
   *size = sizeof(u);
   return (const uint8_t*)&u.buf;